src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -1530,6 +1601,19 @@ GetShardStatistics(MultiConnection *connection, HTAB *shardIds)
 		ShardStatistics *statistics =
 			hash_search(shardStatistics, &shardId, HASH_ENTER, NULL);
 		statistics->totalSize = totalSize;
+
+		if (PQgetisnull(result, rowIndex, 2))
+		{
+			statistics->shardLSN = InvalidXLogRecPtr;
+		}
+		else
+		{
+			char *LSNString = PQgetvalue(result, rowIndex, 2);
+			Datum LSNDatum = DirectFunctionCall1Coll(pg_lsn_in, InvalidOid,
+													 CStringGetDatum(
+														 LSNString));
```

Comment:
If you don't need to specify the colation you can use the DirectFunctionCallX macro instead of the DirectFunctionCallXCol macro:

```suggestion
			Datum LSNDatum = DirectFunctionCall1(pg_lsn_in,
													 CStringGetDatum(
														 LSNString));
```